### ISSUE TITLE
bug with 3.2.2 lint error no-undef defineProps and defineEmits API's

### ISSUE THREAD

**Issue by wcbell51** [vuejs/eslint-plugin-vue#1606]
**Checklist**

- [* ] I have tried restarting my IDE and the issue persists.
- [* ] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:*6.7.2* 
- **eslint-plugin-vue version:*7.0.0* 
- **Node version:*14.15.1* 
- **Operating System:*Windows10Pro* 

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: [
    'plugin:vue/vue3-essential',
    '@vue/airbnb',
    '@vue/typescript/recommended',
  ],
  parserOptions: {
    ecmaVersion: 2020,
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    'max-len': ['error', { code: 150 }],
    'linebreak-style': 0,
  },
};
```

**What did you do?**	

```vue
const props = defineProps({
  name: String,
});
```

**What did you expect to happen?**
-no errors

**What actually happened?**

Module Error (from ./node_modules/eslint-loader/index.js):

  20:15  error  'defineProps' is not defined  no-undef
✖ 1 problem (1 error, 0 warnings)

 @ ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader-v16/dist??ref--0-1!./src/views/Scenes.vue?vue&type=script&setup=true&lang=js 3:0-56 21:19-30
 @ ./src/views/Scenes.vue?vue&type=script&setup=true&lang=js                                                                                                                                                                  tup=true&lang=js 3:0-56 21:19-30       
 @ ./src/views/Scenes.vue
 @ ./src/router/index.ts
 @ ./src/main.ts
 @ multi (webpack)-dev-server/client?http://192.168.86.221:8080&sockPath=/sockjs-node (webpack)/hot/dev-server.js ./src/main.ts

**Repository to reproduce this issue**

-> cannot submit ~ internal repo with my company

thought I'd leave these here as well~ 

![image](https://user-images.githubusercontent.com/5720951/129235299-84b694b8-0787-4981-bb38-7fccce01b65f.png)
screen shot from this page in the docs: 
https://v3.vuejs.org/api/sfc-script-setup.html#using-components

![image](https://user-images.githubusercontent.com/5720951/129235534-cbc6dcba-5828-4848-9c2b-ccf244967f83.png)
this was a post I made in the Vue Land Discord about the issue

![image](https://user-images.githubusercontent.com/5720951/129235658-07f1d875-f501-42bb-bc35-52f66102c38a.png)
this was a DM I had with someone from the vue core team who referred me to post an issue about this in this repository so here I am :)

**Comment by wcbell51** [comment]
hope this helps: right now the only way to get around this is to import defineProps from vue until you all hopefully update the package. thank you in advance!

**Comment by ota-meshi** [comment]
Please read faq.
https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule

**Comment by sodatea** [comment]
Is it possible to only enable these globals in `<script setup>`?

**Comment by ota-meshi** [comment]
There is no way to define `globals` only in `<script setup>`.
You need use `import {defineProps, defineEmits} from 'vue'`, if you don't want to add it to `globals` in other files.

**Comment by KaliaJS** [comment]
@ota-meshi Putting this information in the doc at https://v3.vuejs.org/api/sfc-script-setup.html#defineprops-and-defineemits would be very useful. I looked for a while before I got here.
